### ISSUE TITLE
BraintreePaymentActivity crashes on onBackPressed()

### ISSUE THREAD

**Issue by richk** [braintree/braintree_android#30]
```
05-12 13:46:09.763  13657-13657/org.coursera.android.debug E/AndroidRuntime﹕ FATAL EXCEPTION: main
    Process: org.coursera.android.debug, PID: 13657
    java.lang.NullPointerException: Attempt to invoke virtual method 'void com.braintreepayments.api.Braintree.sendAnalyticsEvent(java.lang.String)' on a null object reference
            at com.braintreepayments.api.dropin.BraintreePaymentActivity.onBackPressed(BraintreePaymentActivity.java:381)
            at android.app.Activity.onKeyUp(Activity.java:2456)
            at android.view.KeyEvent.dispatch(KeyEvent.java:2643)
            at android.app.Activity.dispatchKeyEvent(Activity.java:2707)
            at com.android.internal.policy.impl.PhoneWindow$DecorView.dispatchKeyEvent(PhoneWindow.java:2276)
            at android.view.ViewRootImpl$ViewPostImeInputStage.processKeyEvent(ViewRootImpl.java:4020)
            at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:3982)
            at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:3544)
            at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:3597)
            at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:3563)
            at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:3680)
            at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:3571)
            at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:3737)
            at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:3544)
            at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:3597)
            at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:3563)
            at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:3571)
            at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:3544)
            at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:3597)
            at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:3563)
            at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:3713)
            at android.view.ViewRootImpl$ImeInputStage.onFinishedInputEvent(ViewRootImpl.java:3874)
            at android.view.inputmethod.InputMethodManager$PendingEvent.run(InputMethodManager.java:2208)
            at android.view.inputmethod.InputMethodManager.invokeFinishedInputEventCallback(InputMethodManager.java:1849)
            at android.view.inputmethod.InputMethodManager.finishedInputEvent(InputMethodManager.java:1840)
            at android.view.inputmethod.InputMethodManager$ImeInputEventSender.onInputEventFinished(InputMethodManager.java:2185)
            at android.view.InputEventSender.dispatchInputEventFinished(InputEventSender.java:141)
            at android.os.MessageQueue.nativePollOnce(Native Method)
            at android.os.MessageQueue.next(MessageQueue.java:143)
            at android.os.Looper.loop(Looper.java:122)
            at android.app.ActivityThread.main(ActivityThread.java:5254)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:903)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:698)
```

**Comment by lkorth** [comment]
@richk thanks for the report.

It seems like this is being caused by pressing back before `Braintree` has been initialized. Does this happen when you press back as soon as Drop-in launches and is still on the first loading screen?

**Comment by ramanclabs** [comment]
I'm facing the same issue.
@lkorth yes this happens when we press back as soon as Drop-In launched and is still on the first loading screen.

Please help us out

**Comment by lkorth** [comment]
Thanks for the report @richk, this was fixed in `1.5.1`
